packages/mosaic-tanstack-table-core/src/trimmed/data-table.ts

```diff
@@ -166,32 +167,74 @@ export class MosaicDataTable<
     }
 
     // Add column filters
+    const internalClauses: Array<mSql.FilterExpr> = [];
+
     columnFilters.forEach((columnFilter) => {
       const columnAccessor = this.#columnDefIdToSqlColumnAccessor.get(
         columnFilter.id,
-      )!; // Assertion is safe due to filtering above
-      // Build the filter expression based on the filter value
-
-      // TODO: Support different filter types based on ColumnDef meta.
-      // TODO: Move this building logic into its own utility function/method.
-      // TODO: Figure out the best usage for AND and OR combinations.
-
-      // POC of what this could look like
-      if (columnFilter.value && typeof columnFilter.value === 'string') {
-        // Simple equals filter for now
-        whereClauses.push(
-          mSql.sql`${columnAccessor} ILIKE ${mSql.literal(columnFilter.value.trim())}`,
-        );
+      )!;
+      // Assertion is safe due to filtering above
+
+      // Find the Column Definition to check for metadata
+      const colDef = this.#store.state.columnDefs.find(
+        (c) => c.id === columnFilter.id,
+      );
+
+      const filterType =
+        colDef?.meta?.mosaicDataTable?.sqlFilterType ?? 'equals';
+
+      let clause: mSql.FilterExpr | undefined;
+
+      if (filterType === 'range' && Array.isArray(columnFilter.value)) {
```

Comment:
The `'range'` filter type will need to support instances where the user has defined the `min` or `max` values, but left the opposing value empty. Range should also consider times where the clause will just be checking for `GREATER THAN` or `LESS THAN`.